hs_core/hydroshare/utils.py

```diff
@@ -323,16 +324,35 @@ def get_resource_file_name_and_extension(res_file):
     elif res_file.fed_resource_file_name_or_path:
         f_fullname = res_file.fed_resource_file_name_or_path
 
-    f_fullname = os.path.basename(f_fullname)
+    f_basename = os.path.basename(f_fullname)
     _, file_ext = os.path.splitext(f_fullname)
 
-    return f_fullname, file_ext
+    return f_fullname, f_basename, file_ext
+
+
+def get_resource_file_url(res_file):
+    """
+    Gets the download url of the specified resource file
+    :param res_file: an instance of ResourceFile for which download url is to be retrieved
+    :return: download url for the resource file
+    """
+    if res_file.resource_file:
+        f_url = res_file.resource_file.url
+    elif res_file.fed_resource_file:
+        idx = res_file.fed_resource_file.url.find('/data/contents/')
+        f_url = res_file.fed_resource_file.url[idx + 1:]
+    elif res_file.fed_resource_file_name_or_path:
+        f_url = res_file.fed_resource_file_name_or_path
+    else:
+        f_url = ''
```

Comment:
In normal cases, it will not be possible. But if something goes awry, it could be possible, which we actually encountered during initial backend-frontend integration. In any case, it is always good and will not hurt to include an ```else``` phrase for just-in-case precautions.